c/zssLogging.c

```diff
@@ -43,6 +45,182 @@ bool isLogLevelValid(int level) {
   return TRUE;
 }
 
+typedef struct LogMessagePrefix_tag {
+  char text[LOG_MSG_PREFIX_SIZE];
+} LogMessagePrefix;
+
+typedef struct LogTimestamp_tag {
+  union {
+    struct {
+      char year[4];
+      char delim00[1];
+      char month[2];
+      char delim01[1];
+      char day[2];
+      char delim02[1];
+      char hour[2];
+      char delim03[1];
+      char minute[2];
+      char delim04[1];
+      char second[2];
+      char delim05[1];
+      char hundredth[2];
+    } dateAndTime;
+    char text[22];
+  };
+} LogTimestamp;
+
+
+static void getCurrentLogTimestamp(LogTimestamp *timestamp) {
+
+  uint64 stck = 0;
+  getSTCKU(&stck);
+
+  stck += getLocalTimeOffset();
+
+  stckToLogTimestamp(stck, timestamp);
+
+}
+
+static void getLocationData(char *path, int line, char **locationInfo, uint64 compID, LoggingComponent *component) {
+
+  char prefix[PREFIX_SUFFIX_SIZE]; 
+  char suffix[PREFIX_SUFFIX_SIZE]; 
+
+  unsigned int id = compID & 0xFFFFFFFF;
+  if(compID >= LOG_PROD_COMMON && compID < LOG_PROD_ZIS) {
+    snprintf(prefix,sizeof(LOG_PREFIX_ZCC),LOG_PREFIX_ZCC);
```

Comment:
Since you don't need to do complex formatting, consider using something more lightweight than `snprintf`.